### DISCUSSION TITLE
full selector vs partial selector

### DISCUSSION THREAD

**Discussion by vajrang-b** [vajrang-b/RPA-Developer-in-30-Days#17] [Q&A] [answered]
Describe about full selector and partial selector and write their differences

**Comment by usurabhi** [comment]
full selector : Contain all the elements needed to identify a UI element, including the top-level window, generated by the Basic recorder, Recommended when switching between multiple windows

 partial selector : Generated by the Desktop recorder, Do not contain information about the top-level window, Activities containing partial selectors are enclosed in a container (Attach Browser or Attach Window) that contains a full selector of the top-level window, Recommended when performing multiple actions in the same window

**Comment by TiwariRakesh** [comment]
Full selector- provides the information of UI Elements(Like attributes), basic recorder generate full selector.

partial selector - doesn't keep information of Root node, Desktop recorder uses this selector.

**Comment by Ulhasguttikar** [comment]
full selector :- 1) it contains Root Node with all children nodes till target node.
                       2) Root node is editable in full selector.
                       3) All the UI elements are mapped in full selector that's why it is recommended while switching between multiple windows.

Partial Selector:- 1) Partial Selector of UI element doesn't contain Root Node.
                            2) All the activities containing partial selector are enclosed under container (Attach Browser, Attach Window)
                           3) Recommended while performing multiple operation in one window.

**Comment by mayajain2018** [comment, marked as answer]
**Full Selector:**
1. This selector contains all nodes 
2. Generated by Basic Recording
3. Used while working on many applications
4. Does not have containers
5. Root node can be edited

**Partial Selector:**
1. Root node is inherited (greyed out) and is available only in attach browser/attach window activity
2. Generated by Desktop Recording
3. Used while working on a single application
4. Uses containers
5. Root Node cannot be edited.
